grpc-kotlin/src/test/kotlin/com/linecorp/armeria/server/grpc/kotlin/CoroutineServerInterceptorTest.kt

```diff
@@ -0,0 +1,158 @@
+                assertThat((cause as StatusRuntimeException).status).isEqualTo(Status.UNAUTHENTICATED)
+            })
+    }
+
+    companion object {
+        @RegisterExtension
+        val server: ServerExtension = object : ServerExtension() {
+            override fun configure(sb: ServerBuilder) {
+                val statusFunction = GrpcStatusFunction { _: RequestContext, throwable: Throwable, _: Metadata ->
+                    if (throwable is AnticipatedException && throwable.message == "Invalid access") {
+                        return@GrpcStatusFunction Status.UNAUTHENTICATED
+                    }
+                    // Fallback to the default.
+                    null
+                }
+                val authInterceptor = AuthInterceptor()
+                sb.serviceUnder(
+                    "/non-blocking",
+                    GrpcService.builder()
+                        .exceptionMapping(statusFunction)
+                        .intercept(authInterceptor)
+                        .addService(TestService())
+                        .build()
+                )
+                sb.serviceUnder(
+                    "/blocking",
+                    GrpcService.builder()
+                        .addService(TestService())
+                        .exceptionMapping(statusFunction)
+                        .intercept(authInterceptor)
+                        .useBlockingTaskExecutor(true)
+                        .build()
+                )
+            }
+        }
+
+        private const val username = "Armeria"
+        private const val token = "token-1234"
+
+        private class AuthInterceptor : CoroutineServerInterceptor {
+            private val authorizer = Authorizer { ctx: ServiceRequestContext, _: Metadata ->
+                val future = CompletableFuture<Boolean>()
+                ctx.eventLoop().schedule({
+                    if (ctx.request().headers().contains("Authorization", "Bearer $token")) {
+                        future.complete(true)
+                    } else {
+                        future.complete(false)
+                    }
+                }, 100, TimeUnit.MILLISECONDS)
+                return@Authorizer future
+            }
+
+            override suspend fun <ReqT, RespT> suspendedInterceptCall(
+                call: ServerCall<ReqT, RespT>,
+                headers: Metadata,
+                next: ServerCallHandler<ReqT, RespT>
+            ): ServerCall.Listener<ReqT> {
+                val result = authorizer.authorize(ServiceRequestContext.current(), headers).await()
+                if (result) {
+                    return next.startCall(call, headers)
+                } else {
+                    throw AnticipatedException("Invalid access")
+                }
+            }
+        }
+
+        private class TestService : TestServiceImplBase() {
+            override fun unaryCall(request: SimpleRequest, responseObserver: StreamObserver<SimpleResponse>) {
```

Comment:
- It doesn't look like a stub generated by gRPC-Kotlin. Should we instead use it to test `CoroutineServerInterceptor`?
- The original test suite has a test only for `unaryCall`. Could we add a test for all 4 cases such as `unary`, `client-streaming`, `server-streaming` and `bidi-streaming`?

Reply:
> It doesn't look like a stub generated by gRPC-Kotlin. Should we instead use it to test CoroutineServerInterceptor?

Can you tell me in more detail?

Reply:
I realized that we don't set `kotlin-grpc` flag to this module in `settings.gradle`
`kotlin-grpc` generates a gRPC stub using Kotlin's coroutines that ends with `CoroutineImplBase`.
https://github.com/line/armeria/blob/480321970111ac6249775199fedadcc71b862e84/examples/grpc-kotlin/src/main/kotlin/example/armeria/grpc/kotlin/HelloServiceImpl.kt#L17
https://github.com/line/armeria/blob/480321970111ac6249775199fedadcc71b862e84/settings.gradle#L57

Reply:
I added additional flag in e94328c.
Please double check it cause I'm not sure I did the right adjustment 🙏

Reply:
> The original test suite has a test only for unaryCall. Could we add a test for all 4 cases such as unary, client-streaming, server-streaming and bidi-streaming?

I'm not sure I can do it on proper time because of my noob experience about gRPC & coroutines 😓 
Is there some references in this project? Maybe I can do it with your guidance and reference codes.

Reply:
On the latest commit [fe5b69d](https://github.com/line/armeria/commit/fe5b69dbecb6279b02b5c688da832e4d6852f0c1), I added test cases for 4 methods 🙂